src/satellite/process.ts

```diff
@@ -394,16 +482,32 @@ export class SatelliteProcess implements Satellite {
   }
 
   // Apply a set of incoming transactions against pending local operations,
-  // applying conflict resolution rules. Takes all changes per each key
-  // before merging, for local and remote operations.
-  async _apply(incoming: OplogEntry[], lsn: LSN): Promise<void> {
-    // assign timestamp to pending operations before apply
-    //
+  // applying conflict resolution rules. Takes all changes per each key before
+  // merging, for local and remote operations.
+  async _apply(
+    incoming: OplogEntry[],
+    incoming_origin: string,
+    lsn: LSN
+  ): Promise<void> {
     Log.info(`apply incoming changes for LSN: ${lsn}`)
+    // assign timestamp to pending operations before apply
     await this._performSnapshot()
 
     const local = await this._getEntries()
-    const merged = this._mergeEntries(local, incoming)
+    const merged = this._mergeEntries(
+      this._authState!.clientId,
+      local,
+      incoming_origin,
+      incoming
+    )
+
+    /*
+    console.log(local)
+    console.log(incoming)
+
+    console.log('merged:')
+    console.log(merged)
+     */
```

Comment:
If you don't mind I'm gonna clean it up once we are happy with the review. I left it in case we will need more unit tests. I will wait till I get approval from @balegas on that one.